File: frontend/src/types/SelectOptions.ts
class SelectOptions {
  ignore?: string[];
  searchPattern?: string; //SearchPattern

  take: number = 8;
  skip: number = 0;

  onlyUnlimited?: boolean = true;
  onlyAllAges?: boolean = true;

  favorite?: boolean = false;
  subscribes?: boolean = false;

  orderBy?: SelectOrderBy = SelectOrderBy.None;

  asAdmin?: boolean = false;
}

enum SelectOrderBy {
  None,
  Creation,
  CreationDesc,
  Reports,
  ReportsDesc,
  Views,
}

export default SelectOptions;

export { SelectOrderBy };

File: backend/BYTUBE/Models/SelectOptions.cs
namespace BYTUBE.Models
{
    public class SelectOptions
    {
        public string Ignore { get; set; } = string.Empty;
        public string SearchPattern { get; set; } = string.Empty;

        public int Take { get; set; } = 6;
        public int Skip { get; set; } = 0;

        public bool Favorite { get; set; } = false;
        public bool Subscribes { get; set; } = false;

        public bool AsAdmin { get; set; } = false;

        public SelectOrderBy OrderBy { get; set; } = SelectOrderBy.None;
    }

    public enum SelectOrderBy
    {
        None, Creation, CreationDesc, Reports, ReportsDesc, Views
    }
}
